crates/metaslang/cst/src/lib.rs

```diff
@@ -0,0 +1,42 @@
+pub mod cst;
+pub mod cursor;
+pub mod query;
+pub mod text_index;
+
+pub trait Kind:
+    Sized
+    + Copy
+    + Clone
+    + PartialEq
+    + Eq
+    + std::fmt::Display
+    + std::fmt::Debug
+    + serde::Serialize
+    + for<'a> std::convert::TryFrom<&'a str, Error = strum::ParseError>
+{
+}
+
+impl<T> Kind for T where
+    T: Sized
+        + Copy
+        + Clone
+        + PartialEq
+        + Eq
+        + std::fmt::Display
+        + std::fmt::Debug
+        + serde::Serialize
+        + for<'a> std::convert::TryFrom<&'a str, Error = strum::ParseError>
+{
+}
+
+pub trait TerminalKind: Kind {
+    fn is_trivia(&self) -> bool;
+}
+pub trait NonTerminalKind: Kind {}
+pub trait EdgeKind: Kind {}
+
+pub trait KindTypes: Clone + PartialEq {
```

Comment:
nit:
What do you think of something like `SyntaxKinds` instead of `KindTypes` here, given that it is almost always used as a type parameter `T: KindTypes`?
I'm trying to see if we can avoid using both "kinds" and "types" here if so.

Reply:
`KindTypes` is more accurate, and I was intending to further generify to trees, so not just CSTs. This might be relevant for nanopasses for example

Reply:
Not blocking. We can talk about future plans f2f.